topics/search.md

```diff
@@ -0,0 +1,176 @@
+- For open ended queries, `+inf`, `-inf` can be used to express start and end ranges.
+
+As an example, the following query will return books published between 2021 and 2024 (Both inclusive).
+The equivalent mathematical expression is `2021 <= year <= 2024`.
+
+```
+"@year:[2021 2024]"
+```
+
+While The following query will return books published between 2021 (exclusive) and 2024 (inclusive).
+The equivalent mathematical expression is `2021 < year <= 2024`:
+
+```
+"@year:[(2021 2024]"
+```
+
+The following query will return books published before 2024 (inclusive). The equivalent mathematical expression is year `<= 2024`:
+
+```
+@year:[(-inf 2024]
+```
+
+The following query will return books published after 2015 (exclusive). The equivalent mathematical expression is year `>= 2015`:
+
+```
+@year:[2015 +inf]
+```
+
+### Query planner
+
+A query that utilizes a filter expression to filter results is called a hybrid query.
+Any combination of tag and numeric indexes can form a hybrid query.
+
+- `Pre-filtering`: Pre-filtering relies on secondary indexes (e.g. tag, numeric) to first find the matches to the filter
+ expression regardless of vector similarity. Once the filtered results are calculated a brute-force search is
+ performed to sort by vector similarity.
+- `Inline-filtering`: Inline-filtering performs the vector search algorithm (e.g. HNSW), ignoring found vectors which
+ don't match the filter.
+
+`Pre-filtering` is faster when the filtered search space is much smaller than the original search space. When the
+filtered search space is large, `inline-filtering` becomes faster. The query planner for Valkey-Search automatically
+chooses between the two strategies based on the provided filter.
+
+## Monitoring
+
+To check the server's overall search metrics, you can use the `INFO SEARCH` or `INFO MODULES` commands.
+
+The following metrics are added to the `INFO` command's output:
+
+- `search_used_memory_human`: A human-friendly readable version of the `search_used_memory_bytes` metric
+- `search_used_memory_bytes`: The total bytes of memory that all indexes occupy
+- `search_number_of_indexes`: Index schema total count
+- `search_number_of_attributes`: Total count of attributes for all indexes
+- `search_total_indexed_documents`: Total count of all keys for all indexes
+- `search_background_indexing_status` (String) The status of the indexing process. `NO_ACTIVITY` indicates idle indexing.
+- `search_failure_requests_count`: A count of all failed requests, including syntax errors.
+- `search_successful_requests_count`: A count of all successful requests
+- `search_hnsw_create_exceptions_count`: Count of HNSW creation exceptions.
+- `search_hnsw_search_exceptions_count`: Count of HNSW search exceptions
+- `search_hnsw_remove_exceptions_count`: Count of HNSW removal exceptions.
+- `search_hnsw_add_exceptions_count`: Count of HNSW addition exceptions.
+- `search_hnsw_modify_exceptions_count`: Count of HNSW modification exceptions
```

Comment:
@yairgott Are exceptions the same as errors? My mental model is that an exception is an unhandled internal failure, but I don't know what that would map to here.